pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -709,6 +825,9 @@ func (iqr *IQR) AsResult() (*structs.PipeSearchResponseOuter, error) {
 
 		// Append the known values to the result.
 		for cname, values := range iqr.knownValues {
+			if _, isDeleted := iqr.deletedColumns[cname]; isDeleted {
+				continue
+			}
 			records[cname] = values
 		}
```

Comment:
This is already done in `readAllColumnsWithRRCs()`. So instead of adding to this loop, you should delete the whole loop

Reply:
addressed in the latest commit.